### PR TITLE
electrum: skip cert hostname verification

### PR DESCRIPTION
**Pull request by benma**
In the current setup, the cert is usually self signed, with a missing
CommonName (many public electrum servers work like that). When
connecting a full node, the cert is downloaded and checked manually.

### CODE REVIEW

**Comment by benma** [comment]
@x1ddos please take a look

**Comment by x1ddos** [comment]
Looking. While I'm doing that: why is it needed now. I mean how was it working before this change, or it wasn't? Sorry if this is obvious. Looking at this part of the code base now for the first time.

> with a missing CommonName (many public electrum servers work like that)

Meaning they prefer SANs or that they don't care about names at all?

> the cert is usually self signed

This is my biggest concert. What if it isn't and the user actually wants it to be verified against trusted CA roots like Mozilla & friends?

File: backend/coins/btc/electrum/electrum.go [hunk 1]
```diff
@@ -73,7 +74,40 @@ func newTLSConnection(address string, pemCert string) (*tls.Conn, error) {
...
+			// https://github.com/golang/go/blob/81555cb4f3521b53f9de4ce15f64b77cc9df61b9/src/crypto/tls/handshake_client.go#L327-L344, but adapted to skip the hostname verification.
+			// See https://github.com/golang/go/issues/21971#issuecomment-412836078.
+
+			// If this is the first handshake on a connection, process and
+			// (optionally) verify the server's certificates.
+			certs := make([]*x509.Certificate, len(rawCerts))
+			for i, asn1Data := range rawCerts {
+				cert, err := x509.ParseCertificate(asn1Data)
+				if err != nil {
+					return errp.New("tls: failed to parse certificate from server: " + err.Error())
```
[the remark this answers is not in the document]
Comment (benma) [reply backend/coins/btc/electrum/electrum.go]: good point. `errp` technically wraps everything with a stacktrace, though I am not sure when/if it outputs the stack in an actual failure.

**Comment by benma** [comment]
Yeah, God bless Let's Encrypt :)

Thanks, fixed the naming and error msg prefix. Will leave test for another PR.

**Pull request merged**